src/extension.ts

```diff
@@ -73,14 +73,20 @@ import { Logging } from './shared/logger/commands'
 import { showMessageWithUrl, showViewLogsMessage } from './shared/utilities/messages'
 import { registerWebviewErrorHandler } from './webviews/server'
 import { initializeManifestPaths } from './extensionShared'
+import { ChildProcess } from './shared/utilities/childProcess'
 
 let localize: nls.LocalizeFunc
 
 export async function activate(context: vscode.ExtensionContext) {
     await initializeComputeRegion()
     const activationStartedOn = Date.now()
     localize = nls.loadMessageBundle()
+
     initialize(context)
+    // async init global hostname so it's accessible later to memento
+    const proc = new ChildProcess('hostname', [], { collect: true })
+    globals.hostname = (await proc.run()).stdout.trim() ?? 'unknown-host'
```

Comment:
Should we move this logic in to its own function w/ a different name?
The semantics are something that identifies the "machine", so maybe like "machineId".
And if the `hostname` implementation needs to be changed we only need to update the `getMachineId()` function.
```suggestion
    globals.machineId = getMachineId()
    
    ...

function getMachineId(): string {
  const proc = new ChildProcess('hostname', [], { collect: true })
  await proc.run()).stdout.trim() ?? 'unknown-host'
}
```

Reply:
Got a location in mind?